app/experimenter/experiments/serializers/design.py

```diff
@@ -170,6 +170,21 @@ class Meta(ExperimentDesignVariantBaseSerializer.Meta):
         model = ExperimentVariant
 
     def validate_preferences(self, data):
+        variant = None
+        error_list = [{}] * len(data)
+        for i, pref in enumerate(data):
+            id = pref.get("id")
+            if id and not variant:
+                variant = VariantPreferences.objects.get(id=id).variant
+            if (
+                variant
+                and variant.preferences.filter(pref_name=pref["pref_name"])
+                .exclude(id=pref.get("id"))
+                .exists()
+            ):
+                error_list[i] = {"pref_name": "This pref name already exists."}
+                raise serializers.ValidationError(error_list)
```

Comment:
I think we can simplify this two ways

1) Just do what the other check below this is doing and raise the same error for all the prefs instead of just specific ones

2) You might be able to do the whole thing with a single list comprehension?

```py
swapped_names = [variant.preferences.filter(...).exclude(...).exists() for p in data]
if any(swapped_names):
   raise for all fields
```

Reply:
Would you replace this whole for loop with this ☝️ list comprehension? Right now, `variant` is assigned inside of the for loop, using the id of the first existing preference. Would you do that in a separate step?